EIPS/eip-custom_errors_for_erc_tokens.md

```diff
@@ -0,0 +1,393 @@
+
+Indicates a failure with the `approver` of a token to be approved.
+Used in approvals.
+
+- MUST be used for disallowed approvals from the zero address.
+- MUST NOT be used for transfer operations.
+
+#### `ERC1155InvalidOperator(address operator)`
+
+Indicates a failure with the `operator` to be approved.
+Used in approvals.
+
+- MUST be used for disallowed approvals to the zero address.
+- MUST be used for disallowed approvals to the owner itself.
+- MUST NOT be used for transfer operations.
+  - Use `ERC1155InsufficientApproval` instead.
+
+#### `ERC1155InvalidArrayLength(uint256 idsLength, uint256 valuesLength)`
+
+Indicates an array length mismatch between `_ids` and `_values` in a `safeBatchTransferFrom` operation.
+Used in batch transfers.
+
+- MUST be used only if `idsLength` is different to `valuesLength`
+
+### Parameter Glossary
+
+| Name        | Description                                                                         |
+| ----------- | ----------------------------------------------------------------------------------- |
+| `sender`    | The address of whose token(s) (is/are) transferred from.                            |
+| `balance`   | Current balance for the interacting account.                                        |
+| `needed`    | Amount minimum required to perform an action.                                       |
+| `receiver`  | Address owner of the token(s) transferred.                                          |
+| `spender`   | Address not owner of the token(s) transferred, but allowed to operate on (it/them). |
+| `allowance` | Amount of token(s) a `spender` is allowed to operate with.                          |
+| `approver`  | Owner of the token(s) being approved to an `spender`.                               |
+| `tokenId`   | The identifier number of a token type.                                              |
+| `owner`     | Address of the owner of a token type.                                               |
+| `operator`  | Same as `spender`.                                                                  |
+| `id`        | Same as `tokenId`.                                                                  |
+| `*Length`   | Array length for the prefixed parameter.                                            |
+
+### Error additions
+
+Any addition to this EIP is considerred a new proposal, but SHOULD follow the guidelines presented in the [rationale](#rationale) section to keep consistency.
+
+Implementation specific errors (such as extensions) SHOULD also follow the same guidelines, although subjects and arguments may vary.
+
+## Rationale
+
+The objectives of creating a standard for token errors are to provide context about the error, while keeping gramatical sense, and making a moderated use of meaningful arguments.
+
+Considering the outlined above, the errors are designed based on the standard actions that can be performed on each token, and the [subjects](#actions-and-subjects) involved.
```

Comment:
I'm accepting since I consider your suggestion way better than my original content. But, I'd like to discuss the objectives with you.

My intention with 'keeping grammatical sense" was basically to make the errors _readable_, and not end up with something like `TransferFailedNotOwner(...)`.

The proposed grammar is a direct consequence of this objective. We can keep it as your suggestion but I'd like to consider the readability part of the objectives.